src/adapter/src/coord/statement_logging.rs

```diff
@@ -660,7 +660,13 @@ impl Coordinator {
         params: &Params,
         logging: &Arc<QCell<PreparedStatementLoggingInfo>>,
     ) -> Option<StatementLoggingId> {
-        if session.user().is_internal() {
+        let enable_mz_system_statement_logging = self
+            .catalog()
+            .system_config()
+            .enable_mz_system_statement_logging();
+        if session.user().is_internal()
+            && (!enable_mz_system_statement_logging || !session.user().is_system_user())
```

Comment:
I'd go a step further and make this var about any internal user, not just `mz_system`. I don't think `mz_analytics|mz_monitor|mz_support` will be relevant in a self-hosted context

Reply:
👍🏽 but since we've expanded the scope of the flag can we call it `enable_internal_statement_logging` to make it clear that it's not just `mz_system` that's controlled by the flag?

Reply:
sgtm!